tests/e2e/component.go

```diff
@@ -474,4 +476,166 @@ func componentTests(componentCmdPrefix string) {
 			waitForDeleteCmd("odo project list", projName)
 		})
 	})
+
+	Context("Test odo push with --source and --config flags", func() {
+		//new clean project and context for each test
+		var project string
+		var context string
+
+		//  current directory and project (before eny test is run) so it can restored  after all testing is done
+		var originalDir string
+
+		// Setup up state for each test spec
+		// create new project (not set as active) and new context directory for each test spec
+		// This is before every spec (It)
+		BeforeEach(func() {
+			project = helper.OcCreateRandProject()
+			context = helper.CreateNewContext()
+		})
+
+		// Clean up after the test
+		// This is run after every Spec (It)
+		AfterEach(func() {
+			helper.OcDeleteProject(project)
+			helper.DeleteDir(context)
+		})
+
+		Context("when project flag(--project) is used", func() {
+
+			Context("when using current directory", func() {
+				// we will be testing components that are created from the current directory
+				// switch to the clean context dir before each test
+				JustBeforeEach(func() {
+					originalDir = helper.Getwd()
+					helper.Chdir(context)
+				})
+
+				// go back to original directory after each test
+				JustAfterEach(func() {
+					helper.Chdir(originalDir)
+				})
+
+				It("create local nodejs component and push source and code separately", func() {
+					appName := "nodejs-push-test"
+					cmpName := "nodejs"
+					helper.CopyExample(filepath.Join("source", "nodejs"), context)
+
+					helper.CmdShouldPass("odo component create nodejs " + cmpName + " --app " + appName + " --project " + project)
```

Comment:
Please replace the occurrence of ```odo component``` with ```componentCmdPrefix``` just to make sure component and component sub tests are verified in the new template.